simphony_osp/utils/cache.py

```diff
@@ -0,0 +1,141 @@
+"""Utilities for caching computations."""
+
+from datetime import datetime
+from functools import lru_cache, wraps
+from typing import Any, Callable, Dict, Optional, Tuple
+from weakref import ref
+
+from simphony_osp.utils.other import take
+
+
+def lru_cache_weak(
```

Comment:
Are these decorators tested?

Reply:
They are definitely not broken (as you see they are all around the code in this PR). Should I invest the extra time in adding tests for fine details then? Or are we happy that they work? (e.g. test that memory leaks are indeed prevented).

Reply:
I guess it is not a priority now, but it should probably be done at some point